code/game/objects/items/weapons/switchtool.dm

```diff
@@ -207,3 +215,162 @@
+						"/obj/item/device/flashlight:Light" = null,
+						"/obj/item/weapon/scalpel/laser/tier1:Scalpel" = null)
+
+//Checks the research type and level for the respective field, then adds them all to the stored modules while also filling the slot with that tool.
+/obj/item/weapon/switchtool/holo/attackby(var/obj/item/weapon/D as obj, var/mob/user as mob)
+	if(..())
+		return
+	if(istype(D, /obj/item/weapon/disk/tech_disk))
+		var/obj/item/weapon/disk/tech_disk/T = D
+		var/datum/tech/disk_tech = T.stored
+		if(istype(disk_tech, /datum/tech/biotech) && disk_tech.level >= 3)
+			stored_modules["/obj/item/weapon/circular_saw:Circular saw"] = new /obj/item/weapon/circular_saw(src)
+			stored_modules["/obj/item/weapon/surgicaldrill:Surgical drill"] = new /obj/item/weapon/surgicaldrill(src)
+			stored_modules["/obj/item/weapon/cautery:Cautery"] = new /obj/item/weapon/cautery(src)
+			stored_modules["/obj/item/weapon/hemostat:Hemostat"] = new /obj/item/weapon/hemostat(src)
+			stored_modules["/obj/item/weapon/retractor:Retractor"] = new /obj/item/weapon/retractor(src)
+			stored_modules["/obj/item/weapon/bonesetter:Bonesetter"] = new /obj/item/weapon/bonesetter(src)
+			to_chat(user, "The Holo switchtool has medical designs now!")
+			return
+		if(istype(disk_tech, /datum/tech/engineering) && disk_tech.level >= 3)
+			stored_modules["/obj/item/weapon/screwdriver:Screwdriver"] = new /obj/item/weapon/screwdriver(src)
+			stored_modules["/obj/item/weapon/wrench:Wrench"] = new /obj/item/weapon/wrench(src)
+			stored_modules["/obj/item/weapon/wirecutters:Wirecutters"] = new /obj/item/weapon/wirecutters(src)
+			stored_modules["/obj/item/weapon/crowbar:Crowbar"] = new /obj/item/weapon/crowbar(src)
+			stored_modules["/obj/item/device/multitool:Multitool"] = new /obj/item/device/multitool(src)
+			stored_modules["/obj/item/weapon/weldingtool/experimental:Weldingtool"] = new /obj/item/weapon/weldingtool/experimental(src)
+			to_chat(user, "The Holo switchtool has engineering designs now!")
+			return
+		if(istype(disk_tech, /datum/tech/combat) && disk_tech.level >= 5)
+			stored_modules["/obj/item/weapon/shield/energy:Shield"] = new /obj/item/weapon/shield/energy(src)
+			to_chat(user, "The Holo switchtool has a defensive design now!")
+			return
+		if(istype(disk_tech, /datum/tech/syndicate) && disk_tech.level >= 3)
+			stored_modules["/obj/item/weapon/melee/energy/sword/activated:Sword"] = new /obj/item/weapon/melee/energy/sword/activated(src)
+			to_chat(user, "The Holo switchtool has an offensive design now!")
+			return
+		if(istype(disk_tech, /datum/tech/nanotrasen) && disk_tech.level >= 5)
+			stored_modules["/obj/item/weapon/melee/energy/hfmachete/activated:Sharper sword"] = new /obj/item/weapon/melee/energy/hfmachete/activated(src)
+			to_chat(user, "The Holo switchtool has a secret offensive design now!")
+			return
+	/**
+	//Joke module about power[clean/creep]
```

Comment:
I could actually add it, but it seems a bit silly. I guess an emoji vote could be held. Let's say 5 upboats and a collab being ok with it sound fair? I'm not the one to make the final say though, collabs are.